tests/backup/backup_resiliency_test.go

```diff
@@ -1298,3 +1282,263 @@ var _ = Describe("{RebootNodesWhenBackupsAreInProgress}", func() {
+		})
+		Step("Scaling px-backup deployment replica to 0 and back to original replica while restore is in progress", func() {
+			log.InfoD("Scaling px-backup deployment replica to 0 while restore is in progress")
+			*backupDeployment.Spec.Replicas = scaledDownReplica
+			backupDeployment, err = apps.Instance().UpdateDeployment(backupDeployment)
+			log.Infof("px-backup replica after scaling to 0 is %v", *backupDeployment.Spec.Replicas)
+			log.FailOnError(err, fmt.Sprintf("Scaling down px-backup deployment replica to 0"))
+			dash.VerifyFatal(*backupDeployment.Spec.Replicas == scaledDownReplica, true, "Verifying px-backup deployment replica after scaling down")
+			log.InfoD("Sleeping for 10 minute while restore is in progress and px-backup deployment's replica count is 0")
+			time.Sleep(10 * time.Minute)
+			log.InfoD("Scaling px-backup deployment to original replica while restore is in progress")
+			backupDeployment, err = apps.Instance().GetDeployment(pxBackupDeployment, pxBackupNS)
+			*backupDeployment.Spec.Replicas = originalReplicaCount
+			backupDeployment, err = apps.Instance().UpdateDeployment(backupDeployment)
+			log.FailOnError(err, fmt.Sprintf("Scaling down px-backup deployment replica to original replica count"))
+			log.Infof("px-backup replica after scaling back to original replica is %v", *backupDeployment.Spec.Replicas)
+			dash.VerifyFatal(*backupDeployment.Spec.Replicas == originalReplicaCount, true, "Verify px-backup deployment replica after scaling back to original replica count")
+			log.Infof("Verify that px-backup pod is in Ready state")
+			pxBackupPodStatus := func() (interface{}, bool, error) {
+				backupDeployment, err = apps.Instance().GetDeployment(pxBackupDeployment, pxBackupNS)
+				if err != nil {
+					return "", true, err
+				}
+				if backupDeployment.Status.ReadyReplicas < originalReplicaCount {
+					return "", true, fmt.Errorf("px-backup pod is not ready yet")
+				}
+				return "", false, nil
+			}
+			_, err = task.DoRetryWithTimeout(pxBackupPodStatus, podStatusTimeOut, podStatusRetryTime)
+			log.FailOnError(err, "Validating if the px_backup pod is ready")
+			log.Infof("Number of px_backup pod in Ready state are %v", backupDeployment.Status.ReadyReplicas)
+			dash.VerifyFatal(backupDeployment.Status.ReadyReplicas == originalReplicaCount, true, "Verifying that px_backup pod is in Ready state")
+		})
+		Step("Check if restore is successful after px_backup deployment is scaled back to original replica", func() {
+			log.InfoD("Check if restore is successful after px_backup deployment is scaled back to original replica")
+			ctx, err = backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, restoreName := range restoreNames {
+				err = restoreSuccessCheck(restoreName, orgID, maxWaitPeriodForRestoreCompletionInMinute*time.Minute, 30*time.Second, ctx)
+				dash.VerifyFatal(err, nil, "Inspecting the restore success for - "+restoreName)
```

Comment:
How about log.FailOnError(err, "Getting the status for restore- "+restoreName) similar to backupSuccessCheck?